portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java

```diff
@@ -9,10 +9,29 @@
 /** Handles requests sent to the /hello URL. Try running a server and navigating to /hello! */
 @WebServlet("/hello")
 public class HelloWorldServlet extends HttpServlet {
+    private int pageViews = 0;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    pageViews++;
     response.setContentType("text/html;");
     response.getWriter().println("<h1>Hello Umair!</h1>");
+    response.getWriter().println("<h1>Page Views</h1>");
+    response.getWriter().println("<p>This page has been viewed " + pageViews + " times.</p>");
+    String textValue = request.getParameter("text-input");
+    response.getWriter().println("You submitted: " + textValue);
+  }
+ public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
[4.6.1](https://google.github.io/styleguide/javaguide.html#s4.6.1-vertical-whitespace): A single blank line always appears _Between_ consecutive members or initializers of a class
```suggestion
  }

  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```